Generate NPM tag within publish-release bash script (#2635)
alphagov/govuk-frontend: docs/releasing/publishing-from-a-support-branch.md

### Change the code

> 1. Find out which major version this release will be targeting, for example: if you're releasing v4.x.x, the major version is version 4. To check out the support branch for that major version, run `git checkout support/<MAJOR VERSION NUMBER>.x`. If the branch does not exist, follow these steps to create it:

Reviewer: ```suggestion
1. Find out which major version this release will be targeting, for example, if you're releasing v4.x.x, the major version is version 4. To check out the support branch for that major version, run `git checkout support/<MAJOR VERSION NUMBER>.x`. If the branch does not exist, follow these steps to create it:
```

  You will now be prompted to continue or cancel.

9. Raise a pull request, with `support/<MAJOR VERSION NUMBER>.x` as the base branch to merge into.

10. Once a developer approves the pull request, merge it into `support/<MAJOR VERSION NUMBER>.x`.

### Publish the release to npm

1. Check out `support/<MAJOR VERSION NUMBER>.x` and pull the latest changes.

2. Sign in to npm (`npm login`), using the npm/govuk-patterns-and-tools team [credentials](https://github.com/alphagov/design-system-team-credentials/tree/main/npm/govuk-patterns-and-tools).

3. Run `npm run publish-release`, which will prompt you to check whether the NPM tag looks as expected.

>   If you're following these instructions, it's likely that you're publishing an out-of-order release which should

Reviewer: ```suggestion
  If you're following these instructions, you're probably publishing an out-of-order release.
```

>   *not* be tagged as the 'latest' release on NPM. In this scenario, the tag should take the format `latest-[major-version-number`, for example: `latest-4`.

Reviewer: ```suggestion
  Do not tag this release as the 'latest' release on npm. Instead, give the tag the format `latest-[major-version-number`, for example,`latest-4`.
```

- Does there need to be a ']' after 'number'?
- What's an 'out-of-order' release? Will the users of this doc know?

Author: > Does there need to be a ']' after 'number'?

Thanks, good spot!

> What's an 'out-of-order' release? Will the users of this doc know?

Very good question. I struggled with how to describe this kind of release. I don't think it's clear from 'out-of-order' but wasn't sure how else to describe it. There's a thread on Slack here about it: https://gds.slack.com/archives/CAF8HTJES/p1653049737575209 Happy to hear any thoughts/suggestions!

I'm also going to change the wording because actually I don't think it is likely that the person will be doing one - it's just a lot more likely than if they were following the 'publishing.md' docs.

>   Enter 'y' to continue. If you think the tag should be different, enter `N` to have the option to set your own NPM tag.

Reviewer: ```suggestion
  Enter `y` to continue. If you think the tag should be different, enter `N` to have the option to set your own npm tag.
```

  You will now be prompted to continue or cancel.

9. Raise a pull request, with `support/<MAJOR VERSION NUMBER>.x` as the base branch to merge into.

10. Once a developer approves the pull request, merge it into `support/<MAJOR VERSION NUMBER>.x`.

### Publish the release to npm

1. Check out `support/<MAJOR VERSION NUMBER>.x` and pull the latest changes.

2. Sign in to npm (`npm login`), using the npm/govuk-patterns-and-tools team [credentials](https://github.com/alphagov/design-system-team-credentials/tree/main/npm/govuk-patterns-and-tools).

3. Run `npm run publish-release`, which will prompt you to check whether the npm tag looks as expected.

  If you're publishing a release for a previous major version, do not tag this release as the 'latest' release on npm. Instead, give the tag the format `latest-[major-version-number]`, for example, `latest-4`.

  Enter `y` to continue. If you think the tag should be different, enter `N` to have the option to set your own npm tag.

> 4. You will now be prompted to continue or cancel the release. Check the details and enter `y` to continue. If something doesn't look right, press `N` to cancel the release.

Reviewer: Sorry, being nitpicky :p (Style guide says negative contractions give some users trouble)

```suggestion
4. You will now be prompted to continue or cancel the release. Check the details and enter `y` to continue. If something does not look right, press `N` to cancel the release.
```

  You will now be prompted to continue or cancel.

9. Raise a pull request, with `support/<MAJOR VERSION NUMBER>.x` as the base branch to merge into.

10. Once a developer approves the pull request, merge it into `support/<MAJOR VERSION NUMBER>.x`.

### Publish the release to npm

1. Check out `support/<MAJOR VERSION NUMBER>.x` and pull the latest changes.

2. Sign in to npm (`npm login`), using the npm/govuk-patterns-and-tools team [credentials](https://github.com/alphagov/design-system-team-credentials/tree/main/npm/govuk-patterns-and-tools).

3. Run `npm run publish-release`, which will prompt you to check whether the npm tag looks as expected.

>   If you're following these instructions, you may be publishing an out-of-order release. Do

Reviewer: Just looked at the [Slack convo](https://gds.slack.com/archives/CAF8HTJES/p1653049737575209)...

Can see the logic for each of the suggestions, but I'd probably lean towards 'a release for the current major version', if that's appropriate. I think with each of the others, a new user of this content might wonder "What does 'backport'/'maintenance release'/'support release' mean?"

Author: @EoinShaughnessy I've just had a brainwave and realised that actually I think the only time someone may not want to tag as "latest" is when they're doing a release for a previous major version... I've adjusted the text to reflect this.
